rdt/transformers/base.py

```diff
@@ -328,6 +330,11 @@ def fit(self, data, column):
             column (str):
                 Column name. Must be present in the data.
         """
+        if isinstance(column, list):
+            column = column[0]
+
+        hash_int = int(hashlib.sha256(column.encode('utf-8')).hexdigest(), 16)
+        self._random_seed = hash_int % ((2 ** 32) - 1)  # maximum value for a seed
```

Comment:
Should we change `self.reset_randomization` to use this seed for the reverse_transform as well? Like
```python
self.set_random_state(self._random_seed, 'reverse_transform')
```

If this is only going to apply to generator type transformers then instead maybe we call this `_generator_random_seed`

Reply:
This implementation simply generates a unique `_random_seed` for each transformer. Besides the `AnonymizedFaker`, no transformer currently uses it. The idea would be that, if we add any more random transformers - _which don't depend on the data_ - they can use this value to solve the issue of every column generating the same results.

If I understand you correctly, you are proposing the more general solution of changing the `INITIAL_X_STATE` seed. As I mentioned in the meeting yesterday, I had some trouble getting this implementation to work (particularly because `AnonymizedFaker` sets its seed directly, so changing the random state doesn't do anything). It's definitely doable, it just seemed unnecessary to make the seed of every state different for distinct transformers when we don't currently have any transformers which would benefit from it (besides possibly the `nan` substitution issue).

Reply:
Can we rename it then. Rn it seems like this attribute is being used for all transformers but it isn't